packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -918,18 +919,21 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             maxOpsSinceLastSummary,
             summarizerClientElectionEnabled,
         );
-        // Create the SummaryManager and mark the initial state
-        this.summaryManager = new SummaryManager(
-            context,
-            this.summarizerClientElection,
-            this.runtimeOptions.summaryOptions.generateSummaries !== false,
-            this.logger,
-            this.runtimeOptions.summaryOptions.initialSummarizerDelayMs,
-        );
-
-        if (this.connected) {
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            this.summaryManager.setConnected(this.context.clientId!);
+        // Only create a SummaryManager if summaries are enabled and we are not the summarizer client
+        if (
+            this.runtimeOptions.summaryOptions.generateSummaries !== false
+            && this.context.clientDetails.type !== summarizerClientType
```

Comment:
Instead, use the `SummarizerClientElection.isClientEligible` function? I know the param is a different type, but it should be fairly easy to make it reusable or pad the param.

Reply:
Took a look at this.  I think it would be a positive change to make the signature something like:

```typescript
public static isClientEligible(client: Immutable<IClient>): boolean
```

since eligibility is based on the `IClient` alone, the rest of the `ITrackedClient` doesn't contribute to eligibility consideration.

However, the context still doesn't have access to an `IClient`, so if we wanted to share then probably the best option would be to change `isClientEligible` to have a signature like:

```typescript
public static clientDetailsPermitElection(details: Immutable<IClientDetails> | undefined): boolean
```

Which works but is maybe starting to get a little unfriendly for the other places it gets called.

Reply:
I'll probably leave this as-is unless you feel strongly since the impact to the other callers of `isClientEligible` is somewhat negative.

Reply:
I think it's fine to keep it as is, but can you just make 2 functions? The inner function compares against `IClientDetails`, and the outer (existing) function just calls that on the tracked client's clientDetails? Then the new code uses the inner function?

Reply:
Done in latest